src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -89,6 +92,23 @@ void UpdateResources()
 
             try
             {
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    var filePermissionOctal = Convert.ToInt32("755", 8); // -rwxr-xr-x
+                    const int EINTR = 4;
+                    int chmodReturnCode = 0;
+
+                    for (int retries = 500; chmodReturnCode == -1 && Marshal.GetLastWin32Error() == EINTR && retries > 0; --retries)
```

Comment:
Why is this limited to 500? Maybe add a comment, or remove the limit.

Reply:
I used the same limit as RetryUtil.

Reply:
RetryUtil doesn't check for a specific IO error. This is specifically for handling `EINTR`.
Maybe we're unsure about the `EINTR` value of `4`, which could be the rationale for this limit.

Reply:
This is a good point. In `Syscalls.c` of liblp etc. (which is used in some base systems), they are running it with `code == -1 && errno == EINTR` conditoin. The reason I added additional retries is because i was not sure about the edgecase when it will run indefinitely.

Perhaps it is practically not possible to stuck in this loop forever. I have removed retries from the condition for now. I would need to look it up if this looks like a concern.

Reply:
@am11, I think this loop is fine. There are several places in the CoreCLR PAL where we loop while the `errno` is `EINTR`